gui/controller_history.h

```diff
@@ -0,0 +1,41 @@
+#ifndef CONTROLLER_HISTORY_H
+#define CONTROLLER_HISTORY_H
+
+#include "../common/generated_libs/network_protocol/network_protocol.pb.h"
+
+#include <deque>
+
+// Maintains a history of recent ControllerStatus-es sufficient
+// for rendering the UI.
+//
+// Non-thread-safe, needs external synchronization.
+class ControllerHistory {
+public:
+  using Instant = std::chrono::time_point<std::chrono::steady_clock>;
+
+  ControllerHistory(std::chrono::milliseconds window) : window_(window) {}
+
+  void Append(Instant gui_now, const ControllerStatus &status) {
+    history_.push_back({gui_now, status});
+    while (true) {
```

Comment:
```
while (!history.empty() && gui_now - std::get<0>(history.front()) > window_) { history.pop_front(); }
```

Doesn't use the snazzy C++17-ism, but still simpler I think?

Reply:
Done.